src/librustc_trans/back/link.rs

```diff
@@ -621,6 +621,67 @@ fn link_natively(sess: &Session, dylib: bool,
     let (pname, mut cmd) = get_linker(sess);
     cmd.env("PATH", command_path(sess));
 
+    let is_lld = pname.ends_with("lld");
```

Comment:
LLD on Windows also has versions called `ld.lld.exe` and `lld-link.exe` which have the flavor hard coded in.